src/commands/commit.ts

```diff
@@ -24,11 +28,35 @@ export async function _commit(parsedSettings: ParsedSettings): Promise<void> {
   if (Number.isNaN(newMigrationNumber)) {
     throw new Error("Could not determine next migration number");
   }
-  const newMigrationFilename =
-    String(newMigrationNumber).padStart(6, "0") + ".sql";
-  const currentMigrationPath = getCurrentMigrationPath(parsedSettings);
-  const body = await fsp.readFile(currentMigrationPath, "utf8");
-  const minifiedBody = pgMinify(body);
+
+  // See if we have a message arg
+  const messageIndex =
+    process.argv.findIndex(arg => arg === "--message" || arg === "-m") + 1;
+
+  // If we do, fetch, and replace any whitespace with '_'
+  const messageFromCommandArgs = messageIndex && process.argv[messageIndex];
+  const messageFileContentsMatch = /--! Title:(.*)/.exec(body);
+  const messageFromFileComment =
+    messageFileContentsMatch && messageFileContentsMatch[1];
+
+  const message = messageFromCommandArgs || messageFromFileComment;
+
+  const sluggifiedMessage = message && message.trim().replace(/\s+/g, "_");
+
+  const newMigrationFilename = message
+    ? String(newMigrationNumber).padStart(6, "0") +
+      "-" +
+      sluggifiedMessage +
+      ".sql"
+    : String(newMigrationNumber).padStart(6, "0") + ".sql";
+  if (!isMigrationFilename(newMigrationFilename)) {
+    throw Error("Could not construct migration filename");
+  }
+  if (!isMigrationFilename(newMigrationFilename)) {
+    throw Error("Could not construct migration filename");
+  }
+  const bodyWithoutTitle = body.replace(/--! Title:.*\n/, "");
+  const minifiedBody = pgMinify(bodyWithoutTitle);
```

Comment:
So you don't think we should keep the unmangled message in the commit? I'd expect it to be part of the header, so we can include emoji, HTTP links, etc.